addons/website/static/src/js/content/redirect.js

```diff
@@ -2,13 +2,33 @@
 import { session } from '@web/session';
 
 document.addEventListener('DOMContentLoaded', () => {
-    const htmlEl = document.documentElement;
-    if (!session.is_website_user && !window.frameElement) {
-        const websiteId = htmlEl.dataset.websiteId;
+    if (session.is_website_user) {
+        return;
+    }
+
+    if (!window.frameElement) {
+        const websiteId = document.documentElement.dataset.websiteId;
         const {pathname, search} = window.location;
         const params = new URLSearchParams(search);
         document.body.innerHTML = '';
 
         window.location.replace(`/web#action=website.website_editor&path=${encodeURIComponent(pathname + '?' + params.toString())}&website_id=${websiteId}`);
+    } else {
+        document.addEventListener('click', (ev) => {
+            const isEditorEnabled = document.body.classList.contains('editor_enable');
+            const {href, host, target} = ev.target;
+            const isNewWindow = target === '_blank';
+            const isInIframe = host === window.location.host;
+
+            if (href && !isEditorEnabled && !isNewWindow && !isInIframe) {
+                window.parent.document.dispatchEvent(new CustomEvent('iframeRedirection', {
+                    detail: {
+                        href,
+                    },
+                }));
+                ev.preventDefault();
+            }
+        });
```

Comment:
Strange stuff I don't think this is the way to go, there might be redirection which occur without click and I don't think this is code we want here anyway but in the client action. Can't you check on beforeunload of the iframe if we should indeed leave the backend? Instead of going through that iframeRedirection event stuff?

Reply:
Unfortunately I can't find the redirecting url on the beforeunload event, nor on the load event. It seems to be hidden for security reasons. As I could not find a way to catch an external redirection (actually, the external redirection url), I tried to get it at the click level, but that condition `!isEditorEnabled && !isNewWindow && !isInIframe` seems hazardous. There may be a better way but for now I don't see how to handle that at the client action level